src/channel_request/channel_manager.rs

```
@@ -0,0 +1,892 @@
+	Response,
+};
+use super::utils::check_if_valid;
+
+const SUPPORTED_SPEC_VERSION: u16 = 1;
+
+#[derive(PartialEq)]
+enum ChannelState {
+	InfoRequested,
+	OrderRequested,
+	PendingSelection,
+	PendingPayment,
+	Ready,
+}
+
+struct CRchannel {
+	channel_id: u128,
+	user_id: u128,
+	counterparty_node_id: PublicKey,
+	state: ChannelState,
+	//supported_version: Option<u8>,
+	lsp_balance_sat: Option<u64>,
+	client_balance_sat: Option<u64>,
+	// confirms_within_blocks: Option<u32>,
+	// channel_expiry_blocks: Option<u32>,
+	announce_channel: bool,
+	order_id: Option<OrderId>,
+	info: Option<ChannelInfo>,
+}
+
+impl CRchannel {
+	pub fn new(id: u128, counterparty_node_id: PublicKey, user_id: Option<u128>) -> Self {
+		Self {
+			channel_id: id,
+			user_id: user_id.unwrap_or(0),
+			counterparty_node_id,
+			state: ChannelState::InfoRequested,
+			// supported_version: None,
+			lsp_balance_sat: None,
+			client_balance_sat: None,
+			//confirms_within_blocks: None,
+			//channel_expiry_blocks: None,
+			//token: None,
+			//created_at: None,
+			//expires_at: None,
+			announce_channel: false,
+			order_id: None,
+			info: None,
+		}
+	}
+}
+
+#[derive(Default)]
+struct PeerState {
+	channels_by_id: HashMap<u128, CRchannel>,
+	request_to_cid: HashMap<RequestId, u128>,
+	pending_orders: HashMap<RequestId, Order>,
+}
+
+impl PeerState {
+	pub fn insert_channel(&mut self, channel_id: u128, channel: CRchannel) {
+		self.channels_by_id.insert(channel_id, channel);
+	}
+
+	pub fn insert_request(&mut self, request_id: RequestId, channel_id: u128) {
+		self.request_to_cid.insert(request_id, channel_id);
+	}
+
+	pub fn get_channel_in_state_for_request(
+		&mut self, request_id: &RequestId, state: ChannelState,
+	) -> Option<&mut CRchannel> {
+		let channel_id = self.request_to_cid.remove(request_id)?;
+
+		if let Some(channel) = self.channels_by_id.get_mut(&channel_id) {
+			if channel.state == state {
```

Mh, IIUC this means we would just remove the channel and fail silently if the channel is in a wrong state? We should probably return an error and at least log this, even though it should never happen.